libs/@blockprotocol/graph/src/stdlib/subgraph/element/entity.ts

```diff
@@ -1,51 +1,62 @@
+import { typedEntries } from "../../../shared.js";
 import { Entity, EntityId, EntityRevisionId } from "../../../types/entity.js";
 import { Subgraph } from "../../../types/subgraph.js";
 import { isEntityVertex } from "../../../types/subgraph/vertices.js";
-import { Timestamp } from "../../../types/temporal-versioning.js";
+import { TimeInterval, Timestamp } from "../../../types/temporal-versioning.js";
+import {
+  intervalContainsTimestamp,
+  intervalOverlapsInterval,
+} from "../../interval.js";
+import { mustBeDefined } from "../../must-be-defined.js";
 
 /**
- * Returns all `Entity`s within the vertices of the subgraph, optionally filtering to only get their latest revisions.
+ * Returns all {@link Entity}s within the vertices of the given {@link Subgraph}, optionally filtering to only get their
+ * latest revisions.
  *
  * @param subgraph
  * @param latest - whether or not to only return the latest revisions of each entity
  */
-export const getEntities = (
-  subgraph: Subgraph,
+export const getEntities = <Temporal extends boolean>(
+  subgraph: Subgraph<Temporal>,
   latest: boolean = false,
-): Entity[] => {
+): Entity<Temporal>[] => {
   return Object.values(
```

Comment:
This outer `Object.values` is pointless – it's recreating the array that the inner `Object.values` returns

Reply:
Again good spot, missed it in refactoring 
[Clarity and remove unnecessary call](https://github.com/blockprotocol/blockprotocol/pull/928/commits/64d3c6c1f2b876a3c2481779ae9cad44fba8d36c)